iphone/Maps/Core/iCloud/SynchronizationFileWriter.swift

```diff
@@ -251,6 +249,23 @@ final class SynchronizationFileWriter {
   }
 }
 
+// MARK: - FileManager + FileReplacing
+private extension FileManager {
+  func replaceFileSafe(at sourceUrl: URL, with targetUrl: URL) throws {
+    guard fileExists(atPath: sourceUrl.path) else {
+      LOG(.debug, "Source file \(sourceUrl.lastPathComponent) doesn't exist. The file will be copied.")
+      try copyItem(at: targetUrl, to: sourceUrl)
+      return
+    }
+    let tmpDirectoryUrl = try url(for: .itemReplacementDirectory, in: .userDomainMask, appropriateFor: sourceUrl, create: true)
```

Comment:
Where is this tmp dir located? Did you print it in logs? Is it monitored by the monitor? Is it on the same volume as bookmarks?

Reply:
> Where is this tmp dir located

the default temporary directory will be used for both the local and cloud files (not in the cloud local directory - ive tested both cases)
`file:///private/var/mobile/Containers/Data/Application/6FCB51D2-BD84-4047-AB16-9B0A0C473A1D/tmp/`

But there is an important thing (from the doc to the replace method):
> you can call the [url(for:in:appropriateFor:create:)](doc://com.apple.documentation/documentation/foundation/filemanager/1407693-url) method, passing [FileManager.SearchPathDirectory.itemReplacementDirectory](doc://com.apple.documentation/documentation/foundation/filemanager/searchpathdirectory/itemreplacementdirectory) as the search path directory, to get a temporary URL on the destination's volume that is suitable for use with this method.

`url(for: .itemReplacementDirectory, in: .userDomainMask, appropriateFor: targetUrl, create: true)`
generates the unique path for the **file replacing** to avoid conflicts with the files that can be temporarily stored in the tmp dir (some user generated content). This is why the result path may looks like:
`file:///private/var/mobile/Containers/Data/Application/6FCB51D2-BD84-4047-AB16-9B0A0C473A1D/tmp/NSIRD_Organic%20Maps%20(Debug)_9hSpHR/My%20Places1.kml`

> Did you print it in logs?

I dont log it because this replacing will be called on the every update (like the c++ does) - it is regular file updating

> Is it monitored by the monitor?

No. Because the tmp folder isn't monitored.

> Is it on the same volume as bookmarks?
Yes. Tested on multiple devices.